### ISSUE TITLE
Using the status page with a private repo

### ISSUE THREAD

**Issue by akoenig** [upptime/upptime#54]
Really awesome idea! Thanks for building it 🙂 🚀 

I have one question though. The frontend (status page) is requesting data fron the GitHub API – in order to fetch `history/summary.json` and the incident comments, etc. This results in a case where `upptime` can't be used within a private repository setting where the status page gets still deployed to GitHub Pages. Do you think that this might be a use-case which `upptime` should / will cover in the future or is it already possible and I'm missing something here? 🙂

**Comment by AnandChowdhary** [comment]
No, it's not possible (at this time) to have a status website with data from a private repository. There is probably a way that we can call the GitHub API on built time and regenerate a static website every time there is a new incident, but it'll be hard to keep track of every new comment on an issue. IMHO, it's just better to call the API since the data will always be real-time, so this is not a planned feature.

tl;dr: Not possible for now, no concrete plans to add this functionality.

> If you don't want to have a publicly-available status website, you don't have to enable publishing. If that is the case, you can also choose to keep your repository private. https://upptime.js.org/docs/get-started

**Comment by AnandChowdhary** [comment]
If someone wants to send a PR for this feature, I'll reopen this issue.

**Comment by akoenig** [comment]
Ah, I see. No worries and thanks for the reply. One option which I have in mind is to have a GitHub-compliant API which proxies the needed requests accordingly.

Just saw that the status-page is using octokit. Not sure if it allows to change the API host though.

**Comment by AnandChowdhary** [comment]
Yes, you can specify the `baseUrl` like so: https://octokit.github.io/rest.js/v18#usage

Maybe I can add a configuration option for the base URL, which you can then use to proxy API requests with a personal access token with readonly access to the repository. What do you think?

**Comment by akoenig** [comment]
That would be superb @AnandChowdhary!! Although, we have to find a way for the Graphs as well, right?

**Comment by akoenig** [comment]
@AnandChowdhary I opened a PR which introduces the configuration 🙂 Due to the fact that the graphs are fetched via raw.githubusercontent.com, I had to make sure that when a different API Base URL has been configured, the respective proxy needs to tunnel those requests as well. Luckily, the same authorization mechanism can be used by passing the PAT as an `Authorization` header.

**Comment by AnandChowdhary** [comment]
This is great, thanks!

**Comment by akoenig** [comment]
@AnandChowdhary You're welcome 🙂

Anything I should have in mind when I want to use it? Just created a fresh `upptime` repository from the template and configured the `apiBaseUrl`. At the moment, it still communicates with `api.github.com`.

**Comment by AnandChowdhary** [comment]
Yes, you're right, since you didn't use gitmoji or conventional commits, the automated semantic release didn't trigger. No worries, I've gone ahead and manually added a new release for you to use. 🙂

New release: https://github.com/upptime/status-page/releases/tag/v1.4.0

**Comment by akoenig** [comment]
Awesome, thanks @AnandChowdhary. Ah okay, sorry. I thought I followed your convention and was wondering why there is no conventional commit message in place, but it pointed out that I just overseen it obviously 😅

**Comment by AnandChowdhary** [comment]
No worries, thanks again for your contribution! 😄

**Comment by AnandChowdhary** [comment]
@akoenig, there's a breaking change you should be aware of: In https://github.com/upptime/upptime/issues/107, we decided that we shouldn't publish status websites for private repos by default. I think this is a better default because if you have all private data, exposing it by default isn't the best idea.

For people who have a private repo and still want to publish their status website, they now have to set this:

```yaml
status-website:
  publish: true
```

**Comment by akoenig** [comment]
@AnandChowdhary Ah great, thanks for the info. 😊

Introducing this attribute makes sense 💪
